adsws/accounts/views.py

```diff
@@ -203,6 +204,111 @@ def post(self):
         return {'message': 'success'}, 200
 
 
+
+class UserInfoView(Resource):
+    """
+    Implements getting user info from session ID or token, it should be limited
+    to internal use only
+    """
+    decorators = [
+        ratelimit.shared_limit_and_check("500/43200 second", scope=scope_func),
+        oauth2.require_oauth('adsws:internal')
+    ]
+
+    def post(self):
```

Comment:
For the record, the POST method may not be necessary after all and we may decide not to include it in the final modification. The GET method is the important one.

Reply:
i think it is worth to merge the two into a POST endpoint; and let this one endpoint handle both situations

1. cookie
2. oauth token

or even 3. user_id